src/main/java/io/strimzi/kafka/metrics/KafkaPrometheusMetricsReporter.java

```diff
@@ -30,42 +30,57 @@ public class KafkaPrometheusMetricsReporter implements MetricsReporter {
 
     private static final Logger LOG = LoggerFactory.getLogger(KafkaPrometheusMetricsReporter.class.getName());
 
+    private final PrometheusRegistry registry;
+
     @SuppressFBWarnings({"UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR"}) // Should be investigated as part of https://github.com/strimzi/metrics-reporter/issues/12
     private KafkaMetricsCollector kafkaMetricsCollector;
     @SuppressFBWarnings({"UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR"}) // Should be investigated as part of https://github.com/strimzi/metrics-reporter/issues/12
     private Optional<HTTPServer> httpServer;
 
+    /**
+     * Constructor
+     */
+    public KafkaPrometheusMetricsReporter() {
+        registry = PrometheusRegistry.defaultRegistry;
+    }
+
+    // for testing
+    KafkaPrometheusMetricsReporter(PrometheusRegistry registry) {
```

Comment:
`PrometheusRegistry` has no methods to clear it. For this reason, I introduced constructors to inject a registry instance. This simplifies testing. 
Same in `YammerPrometheusMetricsReporter`